File: src/main/java/no/nav/domain/pensjon/regler/beregning2011/MinstePensjonsniva.java
package no.nav.domain.pensjon.regler.beregning2011;

import no.nav.domain.pensjon.regler.Merknad;
import no.nav.domain.pensjon.regler.kode.FormelKodeCti;
import no.nav.domain.pensjon.regler.kode.MinstepenNivaCti;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * Kopi av PEN 28/8/2009 - ruller tilbake til PREG domenemodell
 * 
 * @author Ørnulf Moen
 */

public class MinstePensjonsniva implements Serializable {
    

    private double belop;
    private double sats;
    private int benyttetYug;
    private int pro_rata_teller_mnd;
    private int pro_rata_nevner_mnd;
    private double pro_rata_brok;
    private FormelKodeCti formelKode;

    /** Minstepensjonsnivå. Kan være lav, ordinær og forhøyet. Benytter tabellen */
    private MinstepenNivaCti satsType;

    private List<Merknad> merknadListe = new ArrayList<Merknad>();

    private double PREG_belopIkkeProratisert;
    private int faktisk_tt_avtaleland_mnd;

    public MinstePensjonsniva() {
        super();
        setFormelKode(new FormelKodeCti("MPNx"));
    }

}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning2011/MinstePensjonsniva.kt
package no.nav.pensjon.regler.domain.beregning2011

import no.nav.pensjon.regler.domain.Merknad
import no.nav.pensjon.regler.domain.enum.FormelKodeEnum
import no.nav.pensjon.regler.domain.enum.MinstePensjonsnivaSatsEnum
import java.io.Serializable

class MinstePensjonsniva : Serializable {
    var belop = 0.0
    var sats = 0.0
    var benyttetYug = 0
    var pro_rata_teller_mnd = 0
    var pro_rata_nevner_mnd = 0
    var pro_rata_brok = 0.0
    var formelKodeEnum: FormelKodeEnum = FormelKodeEnum.MPNx

    /** Minstepensjonsnivå. Kan være lav, ordinær og forhøyet. Benytter tabellen  */
    var satsTypeEnum: MinstePensjonsnivaSatsEnum? = null
    var merknadListe: MutableList<Merknad> = mutableListOf()
    var faktisk_tt_avtaleland_mnd = 0

    constructor()
    constructor(mpn: MinstePensjonsniva) : super() {
        belop = mpn.belop
        sats = mpn.sats
        benyttetYug = mpn.benyttetYug
        pro_rata_teller_mnd = mpn.pro_rata_teller_mnd
        pro_rata_nevner_mnd = mpn.pro_rata_nevner_mnd
        pro_rata_brok = mpn.pro_rata_brok
        if (mpn.satsTypeEnum != null) {
            satsTypeEnum = mpn.satsTypeEnum
        }
        formelKodeEnum = mpn.formelKodeEnum
        for (merknad in mpn.merknadListe) {
            merknadListe.add(Merknad(merknad))
        }
        faktisk_tt_avtaleland_mnd = mpn.faktisk_tt_avtaleland_mnd
    }
}